airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java

```diff
@@ -89,36 +94,67 @@ public static XminStatus getXminStatus(final JdbcDatabase database) throws SQLEx
     return new XminStatus()
         .withNumWraparound(result.get(NUM_WRAPAROUND_COL).asLong())
         .withXminXidValue(result.get(XMIN_XID_VALUE_COL).asLong())
-        .withXminRawValue(result.get(XMIN_RAW_VALUE_COL).asLong());
+        .withXminRawValue(result.get(XMIN_RAW_VALUE_COL).asLong())
+        .withVersion(2L)
+        .withStateType(StateType.XMIN);
   }
 
-  public static void logFullVacuumStatus(final JdbcDatabase database, final ConfiguredAirbyteCatalog catalog, final String quoteString) {
-    catalog.getStreams().forEach(stream -> {
+  static Map<AirbyteStreamNameNamespacePair, Long> fileNodeForStreams(final JdbcDatabase database,
+                                                                      final List<ConfiguredAirbyteStream> streams,
+                                                                      final String quoteString) {
+    final Map<AirbyteStreamNameNamespacePair, Long> fileNodes = new HashMap<>();
+    streams.forEach(stream -> {
+      final AirbyteStreamNameNamespacePair namespacePair =
+          new AirbyteStreamNameNamespacePair(stream.getStream().getName(), stream.getStream().getNamespace());
+      final long l = fileNodeForStreams(database, namespacePair, quoteString);
+      fileNodes.put(namespacePair, l);
+    });
+    return fileNodes;
+  }
+
+  public static long fileNodeForStreams(final JdbcDatabase database, final AirbyteStreamNameNamespacePair stream, final String quoteString) {
+    try {
+      final String streamName = stream.getName();
+      final String schemaName = stream.getNamespace();
+      final String fullTableName =
+          getFullyQualifiedTableNameWithQuoting(schemaName, streamName, quoteString);
+      final List<JsonNode> jsonNodes = database.bufferedResultSetQuery(
+          conn -> conn.prepareStatement(CTID_FULL_VACUUM_REL_FILENODE_QUERY.formatted(fullTableName)).executeQuery(),
+          resultSet -> JdbcUtils.getDefaultSourceOperations().rowToJson(resultSet));
+      Preconditions.checkState(jsonNodes.size() == 1);
+      final long relationFilenode = jsonNodes.get(0).get("pg_relation_filenode").asLong();
+      LOGGER.info("Relation filenode is for stream {} is {}", fullTableName, relationFilenode);
+      return relationFilenode;
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static List<io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair> streamsUnderVacuum(final JdbcDatabase database,
```

Comment:
Hmmm! Am not sure I understand. The XMIN query doesnt query any specific table. It queries the xmin value from the database. If a table is going under vacuum how would it stop us from getting the xmin value?

Reply:
I gave this line as an example
but even queries before and after are hanging if there's a full vacuum going on.
I'd leave our check still just in case, but it is largely a dead code.